src/Datagrid/DatagridMapper.php

```diff
@@ -72,7 +72,7 @@ public function add(
             $fieldDescription->mergeOptions($filterOptions);
         } elseif (\is_string($name)) {
             if ($this->admin->hasFilterFieldDescription($name)) {
-                throw new \RuntimeException(sprintf('Duplicate field name "%s" in datagrid mapper. Names should be unique.', $name));
+                throw new \InvalidArgumentException(sprintf('Duplicate field name "%s" in datagrid mapper. Names should be unique.', $name));
```

Comment:
I think the argument itself is valid. The problem is triggered by our business rules, so `\UnexpectedValueException` or `DomainException` could be a better fit for this case.

Reply:
I recently added a tool to check I try/catch every \Exception and RuntimeException (but not LogicException) in my project.

This one report error because I don't do
```
try {
    $mapper->add('field')
} catch {
    ...
}
```
or
```
/**
  * @throws \RuntimeException
  */
protected function configureListFields(ListMapper $list)
{
}
```

So I thought about it. Does Sonata expect me to try/catch these exception ? No.

I can change the exception for another, but I think this should be one extending LogicException.
DomainException seems indeed better